### ISSUE TITLE
Feature Request - make status bar transparent on devices with notch/punch hole

### ISSUE THREAD

**Issue by Ciancy28** [K1rakishou/Kuroba-Experimental#905]
Currently, KurobaEx makes the status bar transparent and draws content behind it on devices with a regular screen, such as my Sony Xperia Z5 Compact:    
![Screenshot_20230627-224801_KurobaEx-fdroid](https://github.com/K1rakishou/Kuroba-Experimental/assets/77066322/576b2038-83fa-4406-bb21-5fde8a7c255d)
But on devices with a punch hole camera or notch, such as my Google Pixel 4a, the status bar is just black with no transparency at all 
![Screenshot_20230627-224808_KurobaEx-fdroid](https://github.com/K1rakishou/Kuroba-Experimental/assets/77066322/6f50d7d0-d560-420b-b777-915fac6813be)
I'm not sure if this is intended behaviour or not, or if there's a reason for this behaviour difference between screen types, but I suggest making the status bar transparent on all devices, either by default or as an option.

**Comment by Ciancy28** [comment]
Dear bot
This issue/feature request is still relevant and therefore I wish for it to stay open

**Comment by K1rakishou** [comment]
It should be transparent on all phones. If it isn't transparent for you then your phone doesn't support the official API.

**Comment by Ciancy28** [comment]
> It should be transparent on all phones. If it isn't transparent for you then your phone doesn't support the official API.

It's a Google pixel running stock android, and transparent status bar works on every other app, there's no way it doesn't support the official api
Clearly not a device-side problem, my guess is that the app is programmed to only render up to a certain screen length maybe? So the issue is only present on phones with a high aspect ratio

**Comment by K1rakishou** [comment]
>my guess is that the app is programmed to only render up to a certain screen length maybe?

There is no such API. You just tell the window to allow drawing behind the system content and that's it. Everything else is done by Android itself.

**Comment by Ciancy28** [comment]
I installed the app on appetize.io, which uses an emulated almost 100% AOSP build of android and selected a punch hole notch as display cutout in developer options
![Screenshot_20230811-144306](https://github.com/K1rakishou/Kuroba-Experimental/assets/77066322/12719621-3586-45d8-8be1-5829f6048954)
This is further proof that it's not a problem limited to Google Pixels.   
You could also test on the android studio emulator I think and it would probably be the same, but I don't have access to a computer right now.

**Comment by Ciancy28** [comment]
> This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.

would still like to see this implemented
keep open

**Comment by basketweavr** [comment]
Same issue here on MIUI of all things. MIUI 14 based on Android 13 to be specific. I've used this exact same theme since back when this device was on MIUI 12.5.x, it behaved normally back then. Other apps which change the status bar color like Tachiyomi, Anki or Chromium based browsers work normally. 

![image](https://github.com/K1rakishou/Kuroba-Experimental/assets/147221098/b19f3941-9ef6-4ef5-8ffc-ed34b27ef19a)

![image](https://github.com/K1rakishou/Kuroba-Experimental/assets/147221098/75445978-6b04-4e76-964d-c7fc5304515a)

And I can confirm it's related to the punch-hole cutout settings, since the size of the black bar changed when I altered the simulated notch cutout size in Android's developer options.

**Comment by Ciancy28** [comment]
...was this implemented in one of the recent updates?
The status bar is transparent on my 4a now!

**Comment by K1rakishou** [comment]
No, I didn't change anything. Probably your device got an update.

**Comment by Ciancy28** [comment]
> No, I didn't change anything. Probably your device got an update.

Well the pixel 4a hasn't had an official update for years now and I've been running the same A14 custom rom for the past year.
I just tested a bunch of older versions to make sure this wasn't something to do with the device and it seems that the v1.3.36-release from January is what fixed this, every build before that has a black status bar and every build after that has a transparent status bar.
unintentional change? well I'm happy regardless and hope this doesn't change in the future...
Also thanks for having resumed development by the way.

**Comment by K1rakishou** [comment]
I see. I did update a bunch of libraries in the release, so maybe that's what fixed it.
